plugin/evm/block.go

```diff
@@ -156,7 +186,14 @@ func (b *Block) Accept() error {
 	if err != nil {
 		return fmt.Errorf("failed to create commit batch due to: %w", err)
 	}
-	return vm.ctx.SharedMemory.Apply(batchChainsAndInputs, batch)
+	err = vm.ctx.SharedMemory.Apply(batchChainsAndInputs, batch)
+	if err != nil {
+		return err
+	}
+
+	b.vm.atomicTransactor.reset()
```

Comment:
Reset the transactor for the next block. I am not sure this is correct. My understanding is that blocks are processed sequentially, so this is the right place to clear out any transactions.